### PR TITLE
Add instance of any feature

### LINKED ISSUE #157: instance of Any doesn't work
**Describe the bug**
I can not check if a value is of type `Any` using `instance of`.

**To Reproduce**
Expression `x instance of Any` fails to evaluate. 

**Expected behavior**
`Any` is the upper type of all other types. The expression `x instance of Any` should always return `true` if `x` is not `null`. 

**Environment**
* FEEL engine version: 1.11.2

### PR DESCRIPTION
**Pull request by kaaquist**
## Description

* Add check for instance of Any 

closes #157

### CODE REVIEW

File: src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala [hunk 1]
```diff
@@ -122,7 +122,13 @@ class FeelInterpreter {
...
         withVal(eval(x), x => eval(test)(context + (inputKey -> x)))
       case InstanceOf(x, typeName) =>
-        withVal(eval(x), x => withType(x, t => ValBoolean(t == typeName)))
+        println(
+          s" !!! x: ${x.getClass}, value of x: $x, typeName: ${typeName.getClass}, value of typeName: $typeName :: ${eval(x)}")
+        typeName match {
+          case "Any" => ValBoolean(true)
+          case _ =>
+            withVal(eval(x), x => withType(x, t => ValBoolean(t == typeName)))
+        }
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala:131]: :+1:

File: src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala [hunk 2]
```diff
@@ -122,7 +122,13 @@ class FeelInterpreter {
       case In(x, test) =>
         withVal(eval(x), x => eval(test)(context + (inputKey -> x)))
       case InstanceOf(x, typeName) =>
-        withVal(eval(x), x => withType(x, t => ValBoolean(t == typeName)))
+        println(
+          s" !!! x: ${x.getClass}, value of x: $x, typeName: ${typeName.getClass}, value of typeName: $typeName :: ${eval(x)}")
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala:126]: Please remove the log statement :smile:

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala [hunk 3]
```diff
@@ -227,10 +227,4 @@ class BuiltinNumberFunctionsTest
     eval(" odd(-2)") should be(ValBoolean(false))
   }
 
-  "A even() function" should "return true if number is even" in {
-
-    eval(" even(5) ") should be(ValBoolean(false))
-    eval(" even(2) ") should be(ValBoolean(true))
-  }
-
```
Comment (saig0) [inline src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala:235]: I guess this change came from a different PR.

File: src/test/scala/org/camunda/feel/impl/interpreter/InterpreterExpressionTest.scala [hunk 4]
```diff
@@ -60,6 +60,12 @@ class InterpreterExpressionTest
     eval("x instance of string", Map("x" -> 0)) should be(ValBoolean(false))
   }
 
+  it should "be a instance of Any should always pass" in {
+    eval("x instance of Any", Map("x" -> "yes")) should be(ValBoolean(true))
+    eval("x instance of Any", Map("x" -> 1)) should be(ValBoolean(true))
+    eval("x instance of Any", Map("x" -> true)) should be(ValBoolean(true))
+  }
```
Comment (saig0) [inline src/test/scala/org/camunda/feel/impl/interpreter/InterpreterExpressionTest.scala:67]: :+1: 

Please add one more case for `null instance of Any`.

**Review by saig0 — CHANGES_REQUESTED** [review]
Thank you for your contribution @kaaquist :fish_cake: 

Looks good. Just a bit clean up before I can merge it :+1:

File: src/test/scala/org/camunda/feel/impl/interpreter/InterpreterExpressionTest.scala (hunk 4 above)
Comment (kaaquist) [reply src/test/scala/org/camunda/feel/impl/interpreter/InterpreterExpressionTest.scala]: Added extra test and fixed check for `null`.

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala (hunk 3 above)
Comment (kaaquist) [reply src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala:235]: When I merged my fork with the "new changes to upstream/master, this here test got added two times in my code base.

File: src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala [hunk 5]
```diff
@@ -122,7 +122,11 @@ class FeelInterpreter {
       case In(x, test) =>
         withVal(eval(x), x => eval(test)(context + (inputKey -> x)))
       case InstanceOf(x, typeName) =>
-        withVal(eval(x), x => withType(x, t => ValBoolean(t == typeName)))
+        typeName match {
+          case "Any" if eval(x) != ValNull => ValBoolean(true)
+          case _ =>
+            withVal(eval(x), x => withType(x, t => ValBoolean(t == typeName)))
+        }
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala:129]: We can optimize the expression a bit to evaluate `x` only once. 

```suggestion
        withValOrNull(eval(x), x => {
          typeName match {
            case "Any" if x != ValNull => ValBoolean(true)
            case _ => withType(x, t => ValBoolean(t == typeName))
          }
        })
```

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala (hunk 3 above)
Comment (saig0) [reply src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala:235]: Hmm. :thinking: I'm afraid that the test will be deleted when the PR is merged.

**Review by saig0 — CHANGES_REQUESTED** [review]
Thanks :+1: 

Just one minor improvement.

File: src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala (hunk 5 above)
Comment (kaaquist) [reply src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala:129]: Beautiful 😄

**Comment by kaaquist** [comment]
@saig0  after approving the change.. I now get an error: 
```
[ERROR] /Users/kasper/developer/github.com/kaaquist/feel-scala/src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala:125: too many arguments (2) for method withValOrNull: (x: org.camunda.feel.syntaxtree.Val)org.camunda.feel.syntaxtree.Val
```

Do I need to rebase from mater to get the `withValOrNull` function you proposed to use?

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala (hunk 3 above)
Comment (kaaquist) [reply src/test/scala/org/camunda/feel/impl/builtin/BuiltinNumberFunctionTest.scala]: I have added this here test again.. I don't know why it got lost in the first place. But now it is back.

**Comment by kaaquist** [comment]
Ok, I used `withVal` instead of the proposed `withValOrNull` function. Now it compiles and passed all tests 👍

**Review by saig0 — APPROVED** [review]
Thanks :tada:
